src/main/java/de/julielab/ir/umls/UmlsRelationsProvider.java

```diff
@@ -0,0 +1,116 @@
+package de.julielab.ir.umls;
+
+import de.julielab.ir.cache.CacheAccess;
+import de.julielab.ir.cache.CacheService;
+import de.julielab.java.utilities.FileUtilities;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+public class UmlsRelationsProvider {
+    private static final String DEFAULT_SEPARATOR = "\t";
+    private static final Logger log = LogManager.getLogger();
+    private static UmlsRelationsProvider instance;
+    private final String umlsSynsetFile;
+    private final String separator;
+    private static boolean useCache;
+    private CacheAccess<String, Set<String>> parentsCache;
+    private CacheAccess<String, Set<String>> childrenCache;
+    private static String defaultRelationsFile = "resources/umlsRelations.txt.gz";
+
+    public UmlsRelationsProvider(String umlsSynsetFile, boolean useCache) throws IOException {
```

Comment:
Couldn't this be private as well, to force the singleton?

Reply:
It actually can be. It was only public for the unit tests.
However, the workaround now is that there are static fields which can be set by the tests. But that means they could also be set in production code which should happen. So this is a fragile thing.
The issue is: How to properly configure a singletonf or testing?

Reply:
You can use package-private (don't define the accessor level) and have the tests in the same package. Or, you can extend the class in the test and define a public constructor.